File: Programing_Basics_with_JS/forLoopMoreExercises/08.EqualPairs.js
function equalPairs(input) {
    const n = Number(input[0]);
    const firstSum = Number(input[1]) + Number(input[2]);
    let currentSum = 0;
    let difference = 0;
    let maxDiff = 0;
    let isEqual = false;
   
    if (n === 1) {
      console.log(`Yes, value=${firstSum}`);
    } else {
      for (let i = 3; i <= 2 * n; i += 2) {
        currentSum = +input[i] + +input[i + 1];
        if (firstSum === currentSum) {
          isEqual = true;
        } else {
          difference = Math.abs(+input[i - 2] + +input[i - 1] - currentSum);
        }
        if (difference >= maxDiff) {
          maxDiff = difference;
        }
      }
      if (isEqual === false) {
        console.log(`No, maxdiff=${maxDiff}`);
      } else {
        console.log(`Yes, value=${firstSum}`);
      }
    }
  }

equalPairs(['3', '1', '2', '0', '3', '4', '-1'])
equalPairs(['4', '1', '1', '3', '1', '2', '-2', '0', '0'])
equalPairs(['2', '-1', '0', '0', '-1'])
equalPairs(['2', '1', '2', '2', '2'])
equalPairs(['1', '5', '5'])
equalPairs(['2', '-1', '2', '0', '-1'])
equalPairs([7,
    34,
    -33,
    52,
    12,
    -32,
    32,
    23,
    41,
    7,
    25,
    34,
    23,
    124,
    21])

File: Programming_Basics_with_Python/for_loop_more_exercises/08._equal_pairs.py
import sys

numbers_of_couples = int(input())
num1 = int(input())
num2 = int(input())

first_result = num1 + num2
max_difference = -sys.maxsize
is_equal = True

for n in range(2, numbers_of_couples + 1):
    num3 = int(input())
    num4 = int(input())
    current_result = num3 + num4

    if first_result != current_result:
        is_equal = False
        difference = abs(first_result - current_result)
        first_result = current_result
        if difference > max_difference:
            max_difference = difference

if not is_equal:
    print(f'No, maxdiff={max_difference}')
else:
    print(f'Yes, value={first_result}')